### ISSUE TITLE
Getting different loss when setting loss_weights as one-hot vector

### ISSUE THREAD

**Issue by Golbstein** [keras-team/keras#11662]
Hey. I'm training a complex model which is too hard too create a dummy model here, so sorry about that. However, I'm certain I have no "bugs" and yet I'm getting some awkward results.

My model has 3 outputs, hence I'm using 3 different losses that are weighted manually by me:

```
model.compile(optimizer='adam',  loss = {'1' : 'mse', '2' : 'mae', '3' : 'categorical_crossentropy'], 
              loss_weights = [0, 0, 1])
```

I such weighting my final loss **must** be equal to the 3rd loss, but I'm getting different losses (offset of 5%)

output:
[loss: 207.401, 1_loss: 23.921, 2_loss: 197.473, 3_loss: 207.425]

although loss should be equal to 3_loss.

loss 1 and 2 works perfectly and are equal to the "loss" when I train with them. 
I don't have sample\class weights.

Maybe it's something related to the precision of the categorical_crossentropy? 

I'm using Keras 2.2.4 , float32, epsilon 1e-7, python3.5

**Comment by gabrieldemarmiesse** [comment]
Without a script to reproduce this, debugging is *extremely* hard. Could you try to provide one, even if it's not exactly what you are working on?
